### ISSUE TITLE
[1.19.2] Random crashes on world generation on server

### ISSUE THREAD

**Issue by dan212** [P3pp3rF1y/SophisticatedBackpacks#791]
**Describe the bug**
On world generation server crashes.

**To Reproduce**
1.  Move somewhere to generate new chunks
2. Let new monsters spawn
3. With some certainty, crash will happen

**Expected behavior**
Server should not crash.

**Crash logs**
https://pastebin.com/2ravFZr9
https://pastebin.com/axfK1PpG

**Versions**
3.18.45.819

Note:
This seems to happen for other mods as well that reference anything to do with `.level.random` set of functions. Seems to affect server more often than client. One of more descriptive issues on that bug (although from different mod) - https://github.com/AstralOrdana/Immersive-Weathering/issues/192

**Comment by dan212** [comment]
Seems that updating it to latest versions of mods:

- Backpacks: 1.19.2-3.18.50.849
- Core: 1.19.2-0.5.69.312-alpha

Fixed the issue. However, it seems that that code in question might cause that issue in future, so it's worth to look in to.

**Comment by P3pp3rF1y** [comment]
@dan212 you shouldn't be using alpha releases downloaded from github like you have there for sophisticatedcore, but rather just official ones from curseforge unless you want to specifically test something. 
And why would you say that the code may cause issues in the future? This issues was specifically fixed in the latest backpacks release and I am not aware of anything there that could cause issues in the future?

**Comment by dan212** [comment]
I tend to pull last github package release in such cases. If it was fixed in Curseforge release - I will use it next time I will update instead. At the time of me trying to fix the issue the GitHub package was first thing that I found, and it worked. 

Mind that I am not too familiar with Java in general and Minecraft modding in particular. But I saw that same function that seems to cause issues was referenced in a few other places. And since it seemed to be an issue with multi-threading, i was a bit concerned with it dropping up again somewhere else. If it was all cleared out and I am just overly cautious - accept my apology.
